From 72abd3de668fb9868ad6088590848718cd132d75 Mon Sep 17 00:00:00 2001
From: Wenjing Liu <wenjing.liu@amd.com>
Date: Fri, 10 May 2024 11:25:55 -0400
Subject: drm/amd/display: fix a typo which causes an incorrect ODM combine
 setup
Git-commit: 6b7fd8306efbd406fca9e8d27b2c103f951760c8
Patch-mainline: v6.11-rc1
References: jsc#PED-9898 jsc#PED-10191 jsc#PED-10197 jsc#PED-10226 jsc#PED-10237 jsc#PED-10340 jsc#PED-10852 jsc#PED-11022

[why]
A recent change for ODM combine refactor contains a typo which causes ODM
combine mode programmed incorrectly.

Reviewed-by: George Shen <george.shen@amd.com>
Acked-by: Roman Li <roman.li@amd.com>
Signed-off-by: Wenjing Liu <wenjing.liu@amd.com>
Tested-by: Daniel Wheeler <daniel.wheeler@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Acked-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c b/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c
index d085c9ff9d38..42c3c440ed89 100644
--- a/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c
+++ b/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c
@@ -854,7 +854,7 @@ enum dc_status dcn20_enable_stream_timing(
 
 	opp_cnt = resource_get_opp_heads_for_otg_master(pipe_ctx, &context->res_ctx, opp_heads);
 	for (i = 0; i < opp_cnt; i++)
-		opp_inst[opp_cnt] = opp_heads[i]->stream_res.opp->inst;
+		opp_inst[i] = opp_heads[i]->stream_res.opp->inst;
 
 	odm_slice_width = resource_get_odm_slice_dst_width(pipe_ctx, false);
 	last_odm_slice_width = resource_get_odm_slice_dst_width(pipe_ctx, true);
-- 
2.46.1

